src/Symfony/Component/DependencyInjection/Tests/Compiler/AutowirePassTest.php

```diff
@@ -117,6 +117,23 @@ public function testTypeCollision()
         $pass->process($container);
     }
 
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\RuntimeException
+     * @expectedExceptionMessage Unable to guess argument 1 for Symfony\Component\DependencyInjection\Tests\Compiler\NotGuessableArgument because the type-hinted class "Symfony\Component\DependencyInjection\Tests\Compiler\Foo" is used in more than one services.
+     */
+    public function testTypeNotGuessable()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register('a1', __NAMESPACE__.'\Foo');
+        $container->register('a2', __NAMESPACE__.'\Foo');
+        $aDefinition = $container->register('a', __NAMESPACE__.'\NotGuessableArgument');
+        $aDefinition->setAutowired(true);
+
+        $pass = new AutowirePass();
+        $pass->process($container);
+    }
+
```

Comment:
And, what about sub-classes? What if `Foo extends Bar`, all instances in the container are `Foo`, but I type-hint with `Bar`? Or more concretely: what if all instances in the container are `Symfony\Bridge\Monolog\Logger` and I type-hint `Monolog\Logger`? Does this cover that?